### PR TITLE
Add Module class with preliminary testing

### PR DESCRIPTION
**Pull request by rocketninja7**

### CODE REVIEW

File: src/main/java/seedu/address/model/module/Module.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    // Data fields
+    private final Description description;
+    private final Lecturer lecturer;
+    private final Year yearTaken;
+    private final Semester semesterTaken;
+    private final Grade grade;
+
+    public Module(
+            Name name, ModuleCode moduleCode, Description description, Lecturer lecturer,
```
Comment (ji-just-ji) [inline src/main/java/seedu/address/model/module/Module.java:18]: Lecturer into List<Lecturer> for multiple lecturers

File: src/main/java/seedu/address/model/module/Module.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            Year yearTaken, Semester semesterTaken, Grade grade) {
+        requireAllNonNull(name, moduleCode, description, lecturer, yearTaken, semesterTaken, grade);
```
Comment (ji-just-ji) [inline src/main/java/seedu/address/model/module/Module.java:20]: null grade for still taking?

File: src/main/java/seedu/address/model/module/Year.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
+package seedu.address.model.module;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class Year {
+    public static final String MESSAGE_CONSTRAINTS = "Year should only contain 4 digits.";
```
Comment (ji-just-ji) [inline src/main/java/seedu/address/model/module/Year.java:7]: Can we change to Year only takes one digit from year 1 to year 6 instead?

**Review by ji-just-ji — CHANGES_REQUESTED** [review]
Lecturer -> List<Lecturer>
Null/W grade option
Year to 0-6

**Comment by rocketninja7** [comment]
> Lecturer -> List Null/W grade option Year to 0-6

Thanks for the feedback! Just updated the PR with a commit to fix those issues!

**Comment by rocketninja7** [comment]
#67

**Pull request merged**